roles/openshift_preflight/common/tasks/main.yml

```diff
@@ -0,0 +1,22 @@
+---
+# check content available on all hosts
+- when:
+    - not containerized
+  block:
+
+    - name: determine if yum update will work
+      action: check_yum_update
+      register: r
+
+    - set_fact:
+        oo_preflight_check_results: "{{ oo_preflight_check_results + [r|combine({'_task': 'check content available on all hosts'})] }}"
```

Comment:
I'm quite new to Ansible. I feel it is hard to use the word *reliable* and this type of "code" together, it feels like a hack to me :)
There are many ways to introduce subtle mistakes.

From a programming stand point, I think the current approach is already quite crazy and hard to maintain... But was the best we've got working with Ansible so far, after experimenting with different things.

Having different variable names for each role means that introducing a new check requires changes in at least two different places -- the new role and changing the code that checks these variables at the end of the playbook run. And it would be very very easy to miss this second part and assume things are working.